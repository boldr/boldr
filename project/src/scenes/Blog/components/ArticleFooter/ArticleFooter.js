/* @flow */
import React from 'react';
import styled from 'styled-components';
import Link from 'react-router-dom/Link';

export type Props = {
  firstName: string,
  avatarUrl: string,
  lastName: string,
  bio: string,
  username: string,
};

const AFooter = styled.footer`
  @media (min-width: 700px) {
    padding: 70px 100px 0;
  }
`;

const AuthorCardLink = styled.div`
  display: flex;
  align-items: center;
`;

const AuthorImage = styled.img`
  width: 100px;
  height: 100px;
  margin-right: 15px;
  border-radius: 50%;
`;

const AuthorCardContent = styled.section``;

const AuthorName = styled.h4`
  font-size: 20px;
  line-height: 24px;
  font-weight: 600;
  margin: 0 0 5px;
`;

const AuthorBio = styled.p`
  margin: 0;
  color: #738a94;
  font-size: 13px;
  line-height: 18px;
  font-weight: 300;
  @media (min-width: 700px) {
    font-size: 18px;
    line-height: 22px;
  }
`;
const ArticleFooter = ({ avatarUrl, firstName, lastName, username, bio }: Props) => {
  return (
    <AFooter>
      <AuthorCardLink>
        <AuthorImage src={avatarUrl} />
        <AuthorCardContent>
          <AuthorName>
            <Link to={`/profiles/${username}`}>
              {firstName} {lastName}
            </Link>
          </AuthorName>
          <AuthorBio>{bio}</AuthorBio>
        </AuthorCardContent>
      </AuthorCardLink>
    </AFooter>
  );
};

export default ArticleFooter;
