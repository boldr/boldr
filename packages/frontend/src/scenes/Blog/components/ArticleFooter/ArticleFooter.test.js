import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ArticleFooter from './ArticleFooter';

describe('<ArticleFooter />', () => {
  it('accepts props and renders them.', () => {
    const wrapper = shallow(
      <ArticleFooter
        username="TestUser"
        firstName="Steven"
        lastName="Smith"
        avatarUrl="https://boldr.io/images/logo.png"
        bio="This is a bio"
      />,
    );
    expect(wrapper.instance().props.username).toBe('TestUser');
  });
  it('renders snapshot', () => {
    const wrapper = shallow(
      <ArticleFooter
        username="TestUser"
        firstName="Steven"
        lastName="Smith"
        avatarUrl="https://boldr.io/images/logo.png"
        bio="This is a bio"
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
