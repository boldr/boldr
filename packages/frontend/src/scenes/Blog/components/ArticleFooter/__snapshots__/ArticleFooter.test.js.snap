// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleFooter /> renders snapshot 1`] = `
<styled.footer>
  <styled.div>
    <styled.img
      src="https://boldr.io/images/logo.png"
    />
    <styled.section>
      <styled.h4>
        <Link
          replace={false}
          to="/profiles/TestUser"
        >
          Steven
           
          Smith
        </Link>
      </styled.h4>
      <styled.p>
        This is a bio
      </styled.p>
    </styled.section>
  </styled.div>
</styled.footer>
`;
